astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,380 @@
+                'the use of local time scales should be restricted to alternate '
+                'coordinates.'.format(global_info['scale']))
+
+
+def _verify_column_info(global_info, time_col):
+    """
+    Given the column-specific time reference frame information, verify that
+    each column-specific time coordinate attribute has a valid value.
+    Return True if the coordinate column is time, or else return False.
+
+    Parameters
+    ----------
+    global_info : dict
+        Global time reference frame information.
+    time_col : dict
+        Column time reference frame information.
+    """
+
+    scale = time_col.get('scale', None)
+    unit = time_col.get('unit', None)
+
+    if scale is not None:
+
+        if scale.lower() in Time.SCALES:
+            time_col['scale'] = scale.lower()
+            return True
+
+        if scale in FITS_DEPRECATED_SCALES.keys():
+            time_col['scale'] = FITS_DEPRECATED_SCALES[scale]
+            return True
+
+        if scale == 'TIME':
+            time_col['scale'] = global_info['scale']
+            return True if time_col['scale'] else False
+
+        if scale in ['GPS', 'LOCAL']:
+            warnings.warn(
+                'Table column {} has a FITS recognized time scale value {}. '
+                'However, since it is not a valid astropy time scale, '
+                'it will not be converted to astropy Time.'
+                .format(time_col, scale), AstropyUserWarning)
+            return False
+
+        # Non-linear coordinate types have "4-3" form and are not time coordinates
+        if TCTYP_RE_TYPE.match(scale[:5]) and TCTYP_RE_ALGO.match(scale[5:]):
+            return False
+
+        # Coordinate type is either an unrecognized local time scale
+        # or a linear coordinate type
+        return False
+
+    if global_info['scale'] is None:
+        return False
+
+    if (unit is not None and unit in FITS_TIME_UNIT) or 'pos' in time_col:
+        time_col['scale'] = global_info['scale']
+        return True
+
+    return False
+
+
+def _convert_global_time(table, global_info):
+    """
+    Convert the table metadata for time informational keywords
+    to astropy Time.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        The table whose time columns are to be converted.
+    global_info : dict
+        Global time reference frame information.
+    """
+
+    # Read in Global Informational keywords as Time
+    for key, value in global_info.items():
+        # FITS uses a subset of ISO-8601 for DATE-xxx
+        if key.startswith('date'):
+            if key not in table.meta:
+                precision = (lambda x: len(x.split('.')[1])
+                             if '.' in x else 0)
+                scale = 'utc' if key == 'date' else global_info['scale']
+                if scale is not None:
+                    try:
+                        table.meta[key.upper()] = Time(value, scale=scale,
```

Comment:
And to expand on this, the problem is that the keyword will not round trip by default without keeping the precision like this.  E.g. if you read in `1999-01-01T00:00:00.90789` and then write back, it will go in the file as `1999-01-01T00:00:00.908`.